### PR TITLE
Issue 303

### PR DESCRIPTION
**Pull request by twiessner**
Changed implementation for "import.path" config using the spring "Resource" interface. This allows to load not only local directories but also files via HTTP protocol (Basic authentication is also supported). In addition, ZIP files can also be used locally or remotely.

### CODE REVIEW

File: src/main/java/de/adorsys/keycloak/config/provider/FileUtils.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        Collection<File> result = new ArrayList<>();
+        try {
+            ZipFile zip = new ZipFile(zipFile);
+            Enumeration<? extends ZipEntry> entries = zip.entries();
+
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                if (!entry.isDirectory()) {
+                    InputStream inputStream = zip.getInputStream(entry);
+                    result.add(createTempFile(entry.getName(), inputStream));
```
Comment (jkroepke) [inline src/main/java/de/adorsys/keycloak/config/provider/FileUtils.java:84]: Please verify that we are not hitting a Zip Slip Vulnerability

* https://blog.doubleslash.de/zip-slip-gefahr-beim-entpacken-von-zipdateien-mit-java/
* https://snyk.io/research/zip-slip-vulnerability
* https://snyk.io/blog/behind-the-disclosure-the-zip-slip-vulnerability/

File: src/main/java/de/adorsys/keycloak/config/provider/FileUtils.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            default:
+                return Arrays.asList(src);
+        }
+    }
+
+    public static File createTempFile(String name, InputStream inputStream) throws IOException {
+        Assert.notNull(name, "The name of the file to create must be not null!");
+
+        String[] splitted = name.split(REGEX_FILE_NAME_EXTENSION_SPLITTER);
+        Path tempFile = Files.createTempFile(splitted[0], "." + splitted[1]);
```
Comment (jkroepke) [inline src/main/java/de/adorsys/keycloak/config/provider/FileUtils.java:64]: Are the tmp files deleted after running keycloak-config-cli?

Maybe:
```suggestion
        Path tempFile = Files.createTempFile(splitted[0], "." + splitted[1]);
        tempFile.deleteOnExit();
```

**Review by jkroepke — CHANGES_REQUESTED** [review]

**Comment by jkroepke** [comment]
Look good in general. Could you please drop a line about this in CHANGELOG.md?

File: src/main/java/de/adorsys/keycloak/config/provider/FileUtils.java (hunk 1 above)
Comment (twiessner) [reply src/main/java/de/adorsys/keycloak/config/provider/FileUtils.java:86]: good point, but this vulnerability is not affected here. The ZIP file is not unpacked, but only temporary files with the original content are created from it, and the name is normalized when the file is created. I have written a test for this.

File: pom.xml [hunk 3]
```diff
@@ -370,6 +376,8 @@
                     <compilerArgs>
                         <arg>-Xlint</arg>
                     </compilerArgs>
+                    <source>11</source>
```
Comment (jkroepke) [inline pom.xml:379]: Is Java 11 required now?

**Review by jkroepke — APPROVED** [review]
LGTM. Some code coverage is missing for some error casing, but it's good enough.

Thanks Tom!

**Pull request merged**